Library/Homebrew/extend/os/linux/requirements/java_requirement.rb

```diff
@@ -0,0 +1,31 @@
+require "language/java"
+
+class JavaRequirement < Requirement
+  default_formula "jdk"
+
+  env do
+    next unless @java_home
+    env_java
+    java_home = Pathname.new(@java_home)
+    if (java_home/"include").exist? # Oracle JVM
+      env_oracle("linux")
+    end
+  end
+
+  private
+
+  def preferred_java
```

Comment:
Is it worth preferring this over `JAVA_HOME`? I think regardless this logic could perhaps be shared between platforms as Homebrew won't have a `jdk` formula